src/test/ui/type-alias-impl-trait/multiple-def-uses-in-one-fn.stderr

```diff
@@ -0,0 +1,35 @@
+error: defining use generics `[B, A]` differ from previous defining use
+  --> $DIR/multiple-def-uses-in-one-fn.rs:10:45
+   |
+LL | fn f<A, B: 'static>(a: &'static A, b: B) -> (X<A, B>, X<B, A>) {
+   |                                             ^^^^^^^^^^^^^^^^^^
+   |
+note: previous defining use with different generics `[A, B]` found here
+  --> $DIR/multiple-def-uses-in-one-fn.rs:10:45
+   |
+LL | fn f<A, B: 'static>(a: &'static A, b: B) -> (X<A, B>, X<B, A>) {
+   |                                             ^^^^^^^^^^^^^^^^^^
+
+error[E0277]: the trait bound `&'static B: From<&A>` is not satisfied
+  --> $DIR/multiple-def-uses-in-one-fn.rs:7:16
+   |
+LL | type X<A, B> = impl Into<&'static A>;
+   |                ^^^^^^^^^^^^^^^^^^^^^ the trait `From<&A>` is not implemented for `&'static B`
+   |
+   = note: required because of the requirements on the impl of `Into<&'static B>` for `&A`
+
+error: defining use generics `[B, A]` differ from previous defining use
+  --> $DIR/multiple-def-uses-in-one-fn.rs:10:1
+   |
+LL | fn f<A, B: 'static>(a: &'static A, b: B) -> (X<A, B>, X<B, A>) {
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+note: previous defining use with different generics `[A, B]` found here
+  --> $DIR/multiple-def-uses-in-one-fn.rs:10:1
+   |
+LL | fn f<A, B: 'static>(a: &'static A, b: B) -> (X<A, B>, X<B, A>) {
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: aborting due to 3 previous errors
```

Comment:
This duplication is really odd.. we go through the TAIT logic (where I emit this diagnostic) twice. Once with the function as the body, once with the crate root as the body. I haven't been able to figure out the reason for that yet, but will investigate further.